pkg/scheduler/podtask/port_mapping_test.go

```diff
@@ -1,189 +1,65 @@
-			Ports: []api.ContainerPort{{
-				HostPort: 0,
-			}, {
-				HostPort: 1,
-			}},
-		}},
-	}
-	task, err = New(api.NewDefaultContext(), "", *pod, &mesos.ExecutorInfo{})
-	if err != nil {
-		t.Fatal(err)
-	}
-	mapping, err = wildcardHostPortMapping(task, offer)
-	if err, _ := err.(*PortAllocationError); err == nil {
-		t.Fatal("Expected port allocation error")
-	} else if len(err.Ports) != 0 {
-		t.Fatal("Expected port allocation error for wildcard port")
-	}
-
-	//--
-	offer = &mesos.Offer{
-		Resources: []*mesos.Resource{
-			rangeResource("ports", []uint64{1, 2}),
-		},
-	}
-	mapping, err = wildcardHostPortMapping(task, offer)
-	if err != nil {
-		t.Fatal(err)
-	} else if len(mapping) != 2 {
-		t.Fatal("Expected both ports allocated")
-	}
-	valid := 0
-	for _, entry := range mapping {
-		if entry.ContainerIdx == 0 && entry.PortIdx == 0 && entry.OfferPort == 2 {
-			valid++
-		}
-		if entry.ContainerIdx == 0 && entry.PortIdx == 1 && entry.OfferPort == 1 {
-			valid++
+	for i, test := range []struct {
+		PortMapper
+		api.Container
+		Ranges
+		want []PortMapping
+		err  error
+	}{
+		{"fixed", container(), Ranges{}, []PortMapping(nil), nil},
+		{"fixed", container(1, 2), Ranges{}, nil, &PortAllocationError{pod, []uint64{2, 1}}},
+		{"fixed", container(0, 124, 123), Ranges{{100, 150}}, []PortMapping{
+			{pod, 0, 1, 124}, {pod, 0, 2, 123}}, nil},
+		{"fixed", container(123, 123), Ranges{{100, 150}}, nil,
+			&DuplicateHostPortError{PortMapping{pod, 0, 1, 123}, PortMapping{pod, 0, 0, 123}}},
+		{"wildcard", container(123, 123), Ranges{{100, 150}}, nil,
+			&DuplicateHostPortError{PortMapping{pod, 0, 1, 123}, PortMapping{pod, 0, 0, 123}}},
+		{"wildcard", container(), Ranges{}, []PortMapping(nil), nil},
+		{"wildcard", container(123), Ranges{{1, 1}}, nil, &PortAllocationError{pod, []uint64{123}}},
+		{"wildcard", container(0, 0, 1), Ranges{{1, 10}}, []PortMapping{{pod, 0, 2, 1}, {pod, 0, 0, 2}, {pod, 0, 1, 3}}, nil},
+	} {
+		got, err := test.PortMap(test.Ranges, pod, test.Container)
+		if !reflect.DeepEqual(got, test.want) || !reflect.DeepEqual(err, test.err) {
+			t.Errorf("\ntest #%d: %q\ngot:  (%+v, %#v)\nwant: (%+v, %#v)",
+				i, test.PortMapper, got, err, test.want, test.err)
 		}
 	}
-	if valid < 2 {
-		t.Fatalf("Expected 2 valid port mappings, not %d", valid)
-	}
 }
 
-func TestMappingTypeForPod(t *testing.T) {
-	pod := &api.Pod{
-		ObjectMeta: api.ObjectMeta{
-			Labels: map[string]string{},
-		},
-	}
-	mt := MappingTypeForPod(pod)
-	if mt != HostPortMappingWildcard {
-		t.Fatalf("expected wildcard mapping")
-	}
-
-	pod.Labels[PortMappingLabelKey] = string(HostPortMappingFixed)
-	mt = MappingTypeForPod(pod)
-	if mt != HostPortMappingFixed {
-		t.Fatalf("expected fixed mapping")
-	}
-
-	pod.Labels[PortMappingLabelKey] = string(HostPortMappingWildcard)
-	mt = MappingTypeForPod(pod)
-	if mt != HostPortMappingWildcard {
-		t.Fatalf("expected wildcard mapping")
+func container(hostports ...int) api.Container {
```

Comment:
Hmm, that will make the line width explode in the tests. What about just adding a comment to the function? Also, it's not really returning a fake container, it's returning a real container struct, since this isn't a mock implementation of an interface.